src/com/crossbowffs/quotelock/modules/goodreadsqotd/GoodreadsQuoteModule.java

```diff
@@ -0,0 +1,80 @@
+package com.crossbowffs.quotelock.modules.goodreadsqotd;
+
+import android.content.ComponentName;
+import android.content.Context;
+
+import com.crossbowffs.quotelock.R;
+import com.crossbowffs.quotelock.api.QuoteData;
+import com.crossbowffs.quotelock.api.QuoteModule;
+import com.crossbowffs.quotelock.utils.IOUtils;
+import com.crossbowffs.quotelock.utils.Xlog;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GoodreadsQuoteModule implements QuoteModule {
+    private static final String TAG = GoodreadsQuoteModule.class.getSimpleName();
+
+    @Override
+    public String getDisplayName(Context context) {
+        return context.getString(R.string.module_goodreads_name);
+    }
+
+    @Override
+    public ComponentName getConfigActivity(Context context) {
+        return null;
+    }
+
+    @Override
+    public QuoteData getQuote(Context context) throws IOException {
+
+        String quoteAllText = getRss();
+
+        String regexQuoteText = "(?<=<div>\n).*?(?=\n</div>)";
+        Pattern quoteTextPattern = Pattern.compile(regexQuoteText);
+        Matcher quoteTextMatcher = quoteTextPattern.matcher(quoteAllText);
+
+        String regexQuoteCharacter = "(?<=\">).+?(?=</a>)";
+        Pattern quoteCharacterPattern = Pattern.compile(regexQuoteCharacter);
+        Matcher quoteCharacterMatcher = quoteCharacterPattern.matcher(quoteAllText);
+
+        boolean quoteTextStatus = quoteTextMatcher.find();
+        boolean quoteCharacterStatus = quoteCharacterMatcher.find();
+        Xlog.i(TAG, "Quote Text 提取成功？：%s", quoteTextStatus);
+        Xlog.i(TAG, "Quote Text：%s", quoteTextMatcher.group(0));
+        Xlog.i(TAG, "Quote Character 提取成功？：%s", quoteCharacterStatus);
+        Xlog.i(TAG, "Quote Character ：%s", quoteCharacterMatcher.group(0));
+
+        if (quoteTextStatus | quoteCharacterStatus) {
+            String quoteText = quoteTextMatcher.group(0);
+            String quoteSource = String.format("― %s", quoteCharacterMatcher.group(0));
+            return new QuoteData(quoteText, quoteSource);
+        } else {
+            return  null;
+        }
+    }
+
+    /**
+     * Get the quotes_of_the_day rss feed from the Goodreads.com
+     *
+     * @return the string of xml
+     * @throws IOException
+     */
+    public String getRss() throws IOException {
```

Comment:
This should be refactored to `IOUtils` as a utility function so you can reuse this code in the other modules

```Java
public class IOUtils {
    // ...

    public static String downloadString(String urlString) throws IOException {
        URL url = new URL(urlString);
        HttpURLConnection connection = (HttpURLConnection)url.openConnection();
        int responseCode = connection.getResponseCode();
        if (responseCode == 200) {
            return streamToString(connection.getInputStream());
        } else {
            throw new IOException("Server returned non-200 status code: " + responseCode);
        }
    }
}
```